chef/automate
# deleting nodemanager nodes when missing nodes are deleted

### :nut_and_bolt: Description: What code changed, and why?
When missing infra nodes are deleted the associated nodemanager nodes are also deleted. 

### :chains: Related Resources
https://github.com/chef/automate/issues/616

### :athletic_shoe: How to Build and Test the Change
1. `build components/ingest-service && start_all_services`
1. Add an old CCR with` send_chef_run_example`
1. Update the missing node threshold in the node lifecycle https://a2-dev.test/settings/node-lifecycle to 6 hours. This kicks off the process of updating the missing nodes.
1. Go to the client runs page to ensure the node is missing. 
1. Run the below command to see the node in the nodemanager. 
```
curl -f --insecure -H "api-token: $(get_admin_token)" https://localhost/api/v0/nodes/search -d '{
    "filters": [
      {"key": "manager_id", "values": [""]},
      {"key": "name", "values": ["insights.B"]}
    ]
  }' | jq
```
1. Update the Delete Missing Nodes threshold in the node lifecycle https://a2-dev.test/settings/node-lifecycle to 1 minutes. This kicks off the process of updating the deleting nodes.
1. Go to the client runs page to ensure the node is deleted. 
1. Run the below command to see that the node is deleted in the nodemanager. 
```
curl -f --insecure -H "api-token: $(get_admin_token)" https://localhost/api/v0/nodes/search -d '{
    "filters": [
      {"key": "manager_id", "values": [""]},
      {"key": "name", "values": ["insights.B"]}
    ]
  }' | jq
```

### :white_check_mark: Checklist
